topdown/trace.go

```diff
@@ -44,14 +44,23 @@ const (
 	IndexOp Op = "Index"
 )
 
+// VarMetadata provides some user facing information about
+// a variable in some policy.
+type VarMetadata struct {
+	Name     string        `json:"text"`
+	Location *ast.Location `json:"location"`
+}
+
 // Event contains state associated with a tracing event.
 type Event struct {
-	Op       Op            // Identifies type of event.
-	Node     ast.Node      // Contains AST node relevant to the event.
-	QueryID  uint64        // Identifies the query this event belongs to.
-	ParentID uint64        // Identifies the parent query this event belongs to.
-	Locals   *ast.ValueMap // Contains local variable bindings from the query context.
-	Message  string        // Contains message for Note events.
+	Op          Op                     // Identifies type of event.
+	Node        ast.Node               // Contains AST node relevant to the event.
+	Location    *ast.Location          // The location of the Node this event relates to.
+	QueryID     uint64                 // Identifies the query this event belongs to.
+	ParentID    uint64                 // Identifies the parent query this event belongs to.
+	Locals      *ast.ValueMap          // Contains local variable bindings from the query context.
+	VarMetadata map[string]VarMetadata // Contains metadata for the local variable bindings.
```

Comment:
Nit: Can we call this `LocalMetadata` for consistency?

Reply:
Can do, I'll change the json tag to use `local_metadata` to go with it.